File: LeetCode/3370-Smallest-Number-With-All-Set-Bits.cpp
// 3370. Smallest Number With All Set Bits

class Solution {
public:
    int smallestNumber(int n) {
        int ans = 0, bit = 0;
        while(n) {
            ans |= (1 << bit);
            bit++;
            n >>= 1;
        }
        return ans;
    }
};

File: LeetCode/3370-Smallest-Number-With-All-Set-Bits.java
// 3370. Smallest Number With All Set Bits

class Solution {
    public int smallestNumber(int n) {
        int left_most = 0;
        while(n > 0) {
            n >>= 1;
            left_most++;
        }
        return (1 << left_most) - 1;
    }
}
